pd/src/genesis.rs

```diff
@@ -71,79 +63,61 @@ impl FromStr for GenesisAddr {
     }
 }
 
-mod helpers {
+pub mod helpers {
     use decaf377::{FieldExt, Fq};
     use penumbra_crypto::{asset, ka, keys::Diversifier, Value};
 
     use super::*;
 
-    #[derive(Serialize, Deserialize)]
-    pub struct GenesisNotesHelper {
-        notes: Vec<NoteHelper>,
-    }
-
-    impl From<GenesisNotes> for GenesisNotesHelper {
-        fn from(notes: GenesisNotes) -> Self {
-            Self {
-                notes: notes.notes.into_iter().map(From::from).collect(),
-            }
-        }
-    }
-
-    impl TryFrom<GenesisNotesHelper> for GenesisNotes {
-        type Error = anyhow::Error;
-
-        fn try_from(helper: GenesisNotesHelper) -> Result<Self, Self::Error> {
-            Ok(Self {
-                notes: helper
-                    .notes
-                    .into_iter()
-                    .map(TryFrom::try_from)
-                    .collect::<Result<Vec<_>, _>>()?,
-            })
-        }
-    }
-
     #[serde_as]
-    #[derive(Deserialize, Serialize)]
-    pub struct NoteHelper {
-        #[serde_as(as = "serde_with::hex::Hex")]
-        diversifier: [u8; 11],
-        amount: u64,
+    #[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+    pub struct GenesisNote {
         #[serde_as(as = "serde_with::hex::Hex")]
-        note_blinding: [u8; 32],
+        pub diversifier: [u8; 11],
+        pub amount: u64,
         #[serde_as(as = "serde_with::hex::Hex")]
-        asset_id: [u8; 32],
+        pub note_blinding: [u8; 32],
+        pub asset_denom: String,
         #[serde_as(as = "serde_with::hex::Hex")]
-        transmission_key: [u8; 32],
+        pub transmission_key: [u8; 32],
     }
 
-    impl From<Note> for NoteHelper {
-        fn from(note: Note) -> Self {
-            Self {
-                diversifier: note.diversifier().0,
-                amount: note.value().amount,
-                note_blinding: note.note_blinding().to_bytes(),
-                asset_id: note.value().asset_id.to_bytes(),
-                transmission_key: note.transmission_key().0,
-            }
+    impl GenesisNote {
+        pub fn new(
+            diversifier: Diversifier,
+            transmission_key: ka::Public,
+            value: GenesisNoteValue,
+            note_blinding: Fq,
+        ) -> Result<Self, note::Error> {
+            Ok(GenesisNote {
+                diversifier: diversifier.0,
+                amount: value.amount,
+                note_blinding: note_blinding.to_bytes(),
+                asset_denom: value.asset_denom,
+                transmission_key: transmission_key.0,
+                // transmission_key_s: Fq::from_bytes(transmission_key.0)
```

Comment:
can delete, the s-component of the transmission key will be derived when the `Note` is created for realsies